desafio_spring/src/main/java/meli/bootcamp/desafio_spring/services/UserService.java

```diff
@@ -112,6 +118,10 @@ public void unfollowSeller(Long userId, Long sellerId) throws ResourceNotFoundEx
         User user = findUserById(userId);
         Seller seller = sellerService.findSellerById(sellerId);
 
+        if(!sellerRepository.existsByIdAndFollowers_Id(sellerId, userId)){
```

Comment:
Essa validação poderia acontecer antes de ambos findById, para evitar buscas ao banco quando a exceção for lançada.

Reply:
Faz sentido e ja implementei para não ter essa busca desnecessária ao banco. muito obrigado!!